File: Pascal's Triangle/Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        vector<vector<int> > res;
        int count = 0;
        while(count++ < numRows){
            vector<int>     element;
            if(count == 1){
                element.push_back(1);
            }
            else if(count == 2){
                element.push_back(1);
                element.push_back(1);
            }
            else{
                element.push_back(1);
                for(int i = 0; i < count - 2; ++i){
                    element.push_back(res[count - 2][i] + res[count - 2][i + 1]);
                }
                element.push_back(1);
            }
            res.push_back(element);
        }
        return res;
    }
};

File: Pascal's Triangle/Pascal's Triangle.java
public class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ret = new ArrayList<List<Integer>>();
        if(numRows == 0){
            return ret;
        }
        
        while(numRows-- > 0){
            if(ret.size() == 0){
                List<Integer> firstElement = new ArrayList<Integer>();
                firstElement.add(1);
                ret.add(firstElement);
            }
            else{
                List<Integer> prevElement = ret.get(ret.size() - 1);
                List<Integer> nextElement = new ArrayList<Integer>();
                nextElement.add(1);
                for(int i = 0; i < prevElement.size() - 1; ++i){
                    nextElement.add(prevElement.get(i) + prevElement.get(i + 1));
                }
                nextElement.add(1);
                ret.add(nextElement);
            }
        }
        
        return ret;
    }
}
